client/src/index.jsx

```diff
@@ -27,13 +27,14 @@ class ApateezGallery extends React.Component {
     this.clickHandleView = this.clickHandleView.bind(this);
     this.searchRestaurant = this.searchRestaurant.bind(this);
   }
-  componentWillMount(){
-    Modal.setAppElement(document.getElementById('app'));
-  }
+
   componentDidMount(){
-    var id = window.location.href.split('restaurants/')[1];
-    //ajax request for getting the photos and name of restaurant 
-    this.getRequestWithId(id); 
+    if (window) {
+      Modal.setAppElement(document.getElementById('app'));
+      var id = window.location.href.split('restaurants/')[1];
+      //ajax request for getting the photos and name of restaurant
+      this.getRequestWithId(id);
+    }
```

Comment:
Out of curiosity, did the window reference in componentDidMount cause issues for you? AFAIK componentDidMount is not called during server-side rendering but curious if this caused any issues when loading the bundle file server-side.